apps/cowswap-frontend/src/legacy/state/orders/updaters/UnfillableOrdersUpdater.ts

```diff
@@ -185,22 +191,24 @@ export function UnfillableOrdersUpdater(): null {
     strategy,
     updateIsUnfillableFlag,
     isWindowVisible,
-    balances,
     updatePendingOrderPrices,
     verifiedQuotesEnabled,
   ])
 
+  const updatePendingRef = useRef(updatePending)
+  updatePendingRef.current = updatePending
+
   useEffect(() => {
     if (!chainId || !account || !isWindowVisible) {
       console.debug('[UnfillableOrdersUpdater] No need to fetch unfillable orders')
       return
     }
 
     console.debug('[UnfillableOrdersUpdater] Periodically check for unfillable orders')
-    updatePending()
-    const interval = setInterval(updatePending, PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL)
+    updatePendingRef.current()
+    const interval = setInterval(() => updatePendingRef.current(), PENDING_ORDERS_PRICE_CHECK_POLL_INTERVAL)
```

Comment:
updatePending function is the culpit of the re-renders that cause too many queries